logpyle/__init__.py

```diff
@@ -1006,8 +1006,18 @@ def tick_after(self) -> None:
         if self.tick_count + 1 >= self.next_watch_tick:
             self._watch_tick()
 
+        t_step = self.last_values.get("t_step", 0.0)
+        assert isinstance(t_step, float)
+        t_2step = self.last_values.get("t_2step", 0.0)
+        assert isinstance(t_2step, float)
+
         self.t_log += time_monotonic() - tick_start_time
 
+        if (t_step + self.t_log) * 1.1 < t_2step:
+            from warnings import warn
+            warn("warning: t_step + t_log is less than t_2step "
+                 f"{t_step=}, {self.t_log=}, {t_2step=}", stacklevel=2)
```

Comment:
If it matters: I'm not in favor.

- It's ad-hoc (`*1.1`).
- There may be situations where it's not desired.

A better way might be for an application to add a "diagnostics hook" or some such.

Reply:
What do you think of the approach in 653a9ae?

Reply:
Still not a fan. I don't think it's logpyle's place to hardcode (or even suggest) things like this. That logic belongs in the application IMO.

Reply:
Ok, I'll move this code to mirgecom.